src/main/java/org/jenkinsci/test/acceptance/po/PluginManager.java

```diff
@@ -285,7 +285,14 @@ private void tickPluginToInstall(PluginSpec spec) {
         WebElement filterBox = find(By.id("filter-box"));
         filterBox.clear();
         filterBox.sendKeys(name);
-        check(waitFor(by.xpath("//input[starts-with(@name,'plugin.%s.')]", name), 10));
+        WebElement pluginEntry = waitFor(by.xpath("//input[starts-with(@name,'plugin.%s.')]", name), 10);
+        // sometimes the web element gets stale because of the dynamic JS behaviour of the plugin manager UI
+        if (Control.isStale(pluginEntry)) {
```

Comment:
is this actually going to fix the issue described?  
IIUC the staleness is a race condition between the filtering and the resulting display / update and you may not have hit it with this.  (if the element shows up you do the check and *then* the javascript updates the dom)?